### DISCUSSION TITLE
Very slow startup in WSL

### DISCUSSION THREAD

**Discussion by smac89** [litestar-org/litestar#3617] [Potential Issues]
I don't know if this is typical of litestar, or just due to using it through WSL. Whenever I start the dev server (using `litestar run`), it takes a solid minute or more for the server to start up.

I'm running Arch WSL from the Microsoft Store, and this is the timing I get (plus a couple seconds to stop the server) when I run the above command:

```
real    120.01s
user    123.41s
sys     0.40s
```

**2 minutes!**

I also notice that reloading takes a long time as well. Not quite 2 minutes, but definitely longer than it should under normal circumstances. Comparatively speaking, vitejs takes less than half a second to start up, and it is also running in WSL.

Is this normal? What could the problem be?

**Comment by JacobCoffee** [comment]
That's not normal 😬 

Could you share an MCVE/example app or does anything take this long including something like litestar-hello-world (https://github.com/litestar-org/litestar-hello-world)?

**Reply by smac89** [reply]
Interesting! The hello-world application doesn't take any time. Ran it with pdm. I thought maybe it has to do with pdm, so I updated the pyproject file to include poetry config and ran it with poetry, and it was still fast.

Thanks for the suggestion. I think I just have to start all over and figure out where I went wrong

### EDIT
Running the hello-world application using litestar from my project's virtual env also proves to be extremely slow. I did something like `litestar --app-dir /tmp/litestar-hello-world/ run`, and it was back to being slow again. This is after I have wiped my virtual env and rebuilt it again using poetry, which was the same process I used when I used poetry directly with the hello-world project...

**Comment by euri10** [comment]
is it the same with just uvicorn and not using litestar ? did you try another webserver ?

**Reply by smac89** [reply]
> is it the same with just uvicorn and not using litestar ? did you try another webserver ?

Another interesting discovery. Running with just uvicorn starts up instantly. I used `poetry run uvicorn app:app`

**Comment by smac89** [comment]
Turns out the problem is with the `LITESTAR_RELOAD_EXCLUDES` environment variable.

This is what I had in my .env file:

```
LITESTAR_RELOAD_EXCLUDES=.venv/**/*
```

Removing the above line, makes startup instant once more. So the problem looks to be something to do with the way litestar handles ignoring certains files/folders 🙄

**Reply by provinzkraut** [reply]
Can you check if you can reproduce the same behaviour with uvicorn when using its `--reload-exclude?` option`?

That being said, the fact that you have to exclude your venv tells me you might be watching way more files than you're meaning to. What's your reload config?

**Reply by smac89** [reply]
@provinzkraut so these are the variables I have in the .env which affects litestar:

```.env
LITESTAR_HOST=127.0.0.1
LITESTAR_PORT=7000
LITESTAR_DEBUG=true
LITESTAR_RELOAD=true
# https://docs.litestar.dev/2/usage/cli.html#reload-dir
# LITESTAR_RELOAD_EXCLUDES=.venv/**/*
```

With that said, running the application with `poetry run uvicorn app:app --reload-exclude='.venv/**/*'` does not cause any issues with startup speed. It starts up as quickly as before.

_**However,**_ adding the `--reload` flag to the above command causes the startup to hang. I didn't bother waiting for it to continue, but it looks like the same issue resurfaces when reload is enabled.

**Reply by provinzkraut** [reply]
> _**However,**_ adding the `--reload` flag to the above command causes the startup to hang. I didn't bother waiting for it to continue, but it looks like the same issue resurfaces when reload is enabled.

Yeah, so this is an issue with uvicorn. Litestar uses uvicorn's reload capabilities, that's why I'd expected it to be the same with pure uvicorn. You should open an issue over there

**Reply by smac89** [reply]
Discussion started at the uvicorn project. See [here](https://github.com/encode/uvicorn/discussions/2389)
